src/methods/index.js

```diff
@@ -68,8 +68,9 @@ const addAgent = function(NetlifyApi, opts) {
   }
 }
 
-const makeRequestOrRetry = async function(url, opts) {
+const makeRequestOrRetry = async function(url, makeOpts) {
   for (let index = 0; index <= MAX_RETRY; index++) {
+    const opts = makeOpts()
```

Comment:
Instead of creating a function `makeOpts`, then calling it here, I think it would be simpler to:
  - pass `method`, `NetlifyApi`, `requestParams` and `opts` to `makeRequestOrRetry`

```js
const response = await makeRequestOrRetry({ url, method, NetlifyApi, requestParams, opts })
```

  - directly call `getOpts()` here

Reply:
Thanks for the quick feedback, I updated the PR with your suggestions.